.github/workflows/docker-test.yml

```diff
@@ -16,15 +17,32 @@ jobs:
     runs-on: ubuntu-latest
     env:
       SOURCE_ROOT: ${{ github.workspace }}
-      IMAGE: test-image
       ENTRYPOINT: ${{ github.workspace }}/docker/default-entry.sh
     steps:
     - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set variables
+      run: |
+        OWNER_LC=${GITHUB_REPOSITORY_OWNER,,}
+        echo "OWNER_LC=$OWNER_LC" >>${GITHUB_ENV}
+        echo "IMAGE=ghcr.io/$OWNER_LC/vs-fltk:build-env" >>${GITHUB_ENV}
 
     - name: Install dependencies
       run: sudo apt install -y docker-compose
 
+    - name: Check if Dockerfile has changed
+      run: |
+        FILES=$(git log -1 -p ./ | grep +++ | cut -d '/' -f 2-| sed -e 's|dev/null||g')
+        if echo "$FILES" | grep -qE "^(Dockerfile)$"; then
+          echo "changed=true" >> $GITHUB_ENV
+        else
+          echo "changed=false" >> $GITHUB_ENV
+        fi
+
     - name: Build Docker image
+      if: ${{ env.changed == 'true' }}
       run: docker build -t $IMAGE -f docker/Dockerfile ./docker
```

Comment:
I'm experimenting with caching the layers, so that some of the layers might be used in subsequent workflows.

https://github.com/andy5995/linuxdeploy-build-helper-container/pull/21/files

So something like:

```
      - name: Cache Docker layers
        if: ${{ env.changed == 'true' }}
        id: docker-cache
        uses: actions/cache@v3
        with:
          path: ${{ env.DOCKER_CACHE_DEST }}
          key: ${{ runner.os }}-docker-${{ matrix.platform }}-${{ github.ref_name }}
          restore-keys: |
            ${{ runner.os }}-docker-${{ matrix.platform }}-
            ${{ runner.os }}-docker-
      - name: Set up Docker Buildx
        if: ${{ env.changed == 'true' }}
        uses: docker/setup-buildx-action@v3

      - name: Build Docker image
        if: ${{ env.changed == 'true' }}
        run: |
          cd _linuxdeploy_test
          docker buildx build \
            . \
            -t $REGISTRY_IMAGE \
            --build-arg CODENAME=focal \
            --cache-to=type=local,dest=${{ env.DOCKER_CACHE_DEST }} \
            --cache-from=type=local,src=${{ env.DOCKER_CACHE_DEST }} \
            --load
```

If I can get that to work as expected, would you like me to include the changes in this PR?